report_pdf_zip_download/static/src/js/action_manager_report.esm.js

```diff
@@ -0,0 +1,54 @@
+// © 2017 Creu Blanca
+// Copyright 2024 Quartile (https://www.quartile.co)
+// License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+import {download} from "@web/core/network/download";
+import {registry} from "@web/core/registry";
+import {user} from "@web/core/user";
+
+registry
+    .category("ir.actions.report handlers")
+    .add("zip_handler", async function (action, options, env) {
+        if (
+            action.context.active_ids &&
+            action.context.active_ids.length > 1 &&
+            action.report_type === "qweb-pdf" &&
+            action.zip_download === true
+        ) {
+            let url = `/report/zip/${action.report_name}`;
+            const actionContext = action.context || {};
+            if (action.data && JSON.stringify(action.data) !== "{}") {
+                const action_options = encodeURIComponent(JSON.stringify(action.data));
+                const context = encodeURIComponent(JSON.stringify(actionContext));
+                url += `?options=${action_options}&context=${context}`;
```

Comment:
A bit cleaner this way?
```suggestion
            const params = new URLSearchParams();
            if (action.data && JSON.stringify(action.data) !== "{}") {
                params.set("options", JSON.stringify(action.data));
                params.set("context", JSON.stringify(ctx));
                url += `?${params.toString()}`;
```